### DISCUSSION TITLE
b-form-spinbutton update max value

### DISCUSSION THREAD

**Discussion by Joe-Westra** [bootstrap-vue/bootstrap-vue#6596] [Q&A]
I have a grid of `b-form-spinbutton`'s with an initial min = 0  and max = 1 (step of .01).  I want the sum of all spinbuttons to be a max of 1.
The hierarchy looks like this:
```
<CategorizationGrid>
    <PercentagePicker> (array of these)
        <b-form-spinbutton> (one per PercentagePicker)
```
the `<b-form-spinbutton>` has an `@change` listener that emits the difference in value on each change.
The <CategorizationGrid> listens for these events can saves 1 - <the sum of these changes> into `computedMaxVal` which is used as a prop for the <PercentagePicker>.  I have put a `watch` on that prop in the PercentagePicker and can see that the logic is working well, however when setting the `<b-form-inputspinner max:computedMaxValue>` the max on the spinner doesn't change dynamically.  The goal is to remove the users ability to set the sum of the b-form-spinbuttons to more than 1 (or less than 0)
I imagine that I can either:
- Disable the increment or decrement buttons based on the value of 'computedMaxValue' and the spinbuttons current value
    OR
- Change the value in the spinbutton back to a previous value if the computedMaxValue is less than 0

Are there better options?  (ie. re-create the inputbutton with a max value on change and the place focus on it?)

Thanks!

**Comment by Hiws** [comment]
Here's an example of what you're trying to achieve. (to my understanding)
https://codepen.io/Hiws/pen/abJoLgQ

The above uses two computed properties to calculate the current sum and how much is remaining based on the `maxSum` prop.
I then use the `remaining` computed prop in the `max` prop on `<b-spinbutton>` to set the max value for each spin button.

**Reply by Joe-Westra** [reply]
Thanks for your prompt response and a working sample!  This is most encouraging.  Unfortunately, my needs are an abstraction away from the sample that you provided and I don't currently understand Vue enough to implement it as the max value for the spin button in my project isn't being respected.  You have shown me that it should be possible, and I'll take this as an opportunity to learn more :)
Thank you!

**Reply by Hiws** [reply]
If you can create a minimal reproduction of your current code/setup, i can take another look to see.
Other wise hope you find the fix 😄
